src/common/transformations/src/ov_ops/fully_connected_compressed.cpp

```diff
@@ -0,0 +1,85 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "ov_ops/fully_connected_compressed.hpp"
+
+#include "matmul_shape_inference.hpp"
+#include "openvino/core/type/element_type.hpp"
+
+namespace ov {
+namespace op {
+namespace internal {
+
+FullyConnectedCompressed::FullyConnectedCompressed(const OutputVector& arguments, const ov::element::Type output_type)
+    : FullyConnected(OutputVector(arguments.begin(), arguments.begin() + 3), output_type) {
+    for (size_t i = 3; i < arguments.size(); i++) {
+        set_argument(i, arguments[i]);
+    }
+    validate_and_infer_types();
+}
```

Comment:
To be honest, I don't really like constructors from `OutputVector`. This is not safe enough, since they allow user to pass vector with any size to the constructor, and e.g. if `arguments.size() < 3`, we read out of memory here, so we can get hardly understandable error in runtime.
I understand benefits of this constructor: it allows to make some other class methods shorter. Maybe we can make this constructor not public then? And make sure before each constructor creation that arguments size is applicable (it seems like this is true for all the existing use cases)

Reply:
There is no problem to add additional asserts.
To be honest I don't really see the benefits of having all the arguments passed as a separate parameter.
When they are going to be put into a vector anyway.
And, for example, using the current approach on master, when we clone an operation, we take existing parameters, then pass them one by one, and then again put them into a vector to a new operation.
Such clone should be really lightwell operation, but it is basically impossible to make it lightwell because of this inconsistent design.
Those transfers from one representation into another just does not make sense.
They are more verbose and also slower. Probably the root cause is that underlying ov::Node ov::Op architecture is not really that perfect.
Maybe we should not try to fix this in scope of the current PR.
Another alternative would be to just make the most verbose version, when we just always require all the arguments to be passed explicitly.
Then `clone_with_new_inputs` will just take all the inputs from the vector and pass them one by one into a constructor unconditionally.
@vladimir-paramuzov @slyalin @dmitry-gorokhov @itikhono What do you think?
Any feedback is appreciated.

Reply:
I have reverted all the interface changes to simplify the PR changes
Let's improve it later if we want